### DISCUSSION TITLE
Feature Request: Front Matter/Tagging

### DISCUSSION THREAD

**Discussion by kfb1007** [docmost/docmost#1640] [Ideas]
Thanks for everyone's work on this project. I'm running on pikapods and its been working well. 

It would be great to have some kind of front matter/tagging capability to better organize notes.

Thanks!

**Comment by schittli** [comment]
**According to Tags / Tagging,**
it would be brilliant if one could:
- Assign several tags to individual pages
- Limit the search to pages that have certain tags
- Display all pages for a tag via URL

Thanks a lot, kind regards,
Thomas

**Comment by wixregiga** [comment]
I would also add some default frontmatter like:

- Original Author
- Last Updated 
- Last Updated By

**Comment by HallexCosta** [comment]
> I would also add some default frontmatter like:
> 
> * Original Author
> * Last Updated
> * Last Updated By

@kfb1007 @schittli @wixregiga The default front matter would it be something like this?

![image](https://github.com/user-attachments/assets/a6557fd0-f477-47d1-80cb-c5017977fe56)

**Comment by nie3mals** [comment]
I am also looking for page tagging and limit search based on tags. very use full for new onboarded users.

**Comment by marcvanandel** [comment]
Tip: [SilverBullet](https://silverbullet.md/Frontmatter) is a very interesting project for features like frontmatter. It would be very nice to incorporate frontmatter in a similar way into Docmost 😃

**Comment by wcherry** [comment]
So I'm working on contributing some of this functionality; specifically tags.  
Tags would be:
* Displayed on the top of each page
* Show up in the search dialog and be clickable to limit the search to pages that have that tag
* Free form - any text except for space and comma
* Clicking on a tag would open the search dialog and show all pages with that tag

@kfb1007 and @schittli would that meet your needs for tags? Is there more functionality that you think should be required?

![Image](https://github.com/user-attachments/assets/5c2d7479-0328-401f-9dc5-e3c365644c93)

**Comment by AndrewSav** [comment]
I also second the feature request about tags. @wcherry looking good to me.

**Comment by AndrewSav** [comment]
Bookstack has optional values for tags, e.g. Color=Red instead of simply Color. This gives more flexibility, as you can find all pages tagged with any color, as well as pages tagged with a particular color specifically.

**Comment by dmitrii-smolkin** [comment]
Just chiming in — I think tags would be a really useful feature.

**Comment by marcvanandel** [comment]
Maybe there are multiple requests in this discussion? It seems that it started with the request for _tags_ 'to organise notes', right? Secondly, and within the feature request, a solution has been requested or proposed: _front matter_. And front matter can be used for many other features as well.

But ... tags can be solved in different ways and still front matter would be a great solution for other feature requests as well. So should this discussion be about _tags_? Or should this be about _front matter_? 🤔

**Comment by mustafa-yilmaz** [comment]
User groups could also be tagged in order to manage (track, notify) people with group-based tasks.

**Comment by Philipinho** [comment]
Labels/tags will be available in the next version release.

PR: https://github.com/docmost/docmost/pull/2188.
